olp-cpp-sdk-authentication/src/AuthenticationClientUtils.cpp

```diff
@@ -262,6 +223,41 @@ AuthorizeResult GetAuthorizeResult(rapidjson::Document& doc) {
   return result;
 }
 
+UserAccountInfoResponse GetUserAccountInfoResponse(
+    client::HttpResponse& http_response) {
+  using UserAccountInfo = model::UserAccountInfoResponse;
+
+  rapidjson::Document document;
+  const auto api_error = Parse(http_response, document);
+  if (!api_error.GetMessage().empty()) {
+    return api_error;
+  }
```

Comment:
As you can see this is not the proper way of doing it. Seems you are misusing badly the ApiError class. Return from an API should be always ApiResponse which has the `IsSuccessful()` method to check the outcome of the operation. Why dont you use ApiResponse<rapidjson::Document, ApiError> as return value. Seems rapidjson::Document has move constructor/operator and can be easily movable.